synapse/api/errors.py

```diff
@@ -145,6 +145,22 @@ def error_dict(self):
         return cs_error(self.msg, self.errcode)
 
 
+class ShadowBanError(SynapseError):
```

Comment:
shouldn't this be a CodeMessageException rather than a SynapseError? (it doesn't have a matrix errorcode, for example)

Reply:
hrm. CodeMessageExceptions will get logged at ERROR rather than INFO, which probably makes that wrong (but I am still unconvinced either are correct)

Reply:
This got rewritten to be a generic `Exception`, so I think this point is moot now.